apps/system/js/app_install_manager.js

```diff
@@ -1,714 +1,718 @@
-     'handleDownloadError',
-     'handleProgress',
-     'handleApplicationReady'
-    ].forEach(function(name) {
-      this[name] = this[name].bind(this);
-    }, this);
-
-    window.addEventListener('applicationready',
-        this.handleApplicationReady);
-
-    window.addEventListener('home', this.handleHomeButtonPressed.bind(this));
-  },
-
-  handleHomeButtonPressed: function ai_handleHomeButtonPressed(e) {
-    this.dialog.classList.remove('visible');
-    this.handleInstallCancel();
-
-    // hide IME setup dialog if presented
-    if (this.setupInstalledAppDialog.classList.contains('visible') ) {
-      this.handleSetupCancelAction();
-    }
-
-    // hide IME layout list if presented
-    if (this.imeLayoutDialog.classList.contains('visible')) {
-      this.hideIMEList();
-    }
-  },
-
-  handleApplicationReady: function ai_handleApplicationReady(e) {
-    window.removeEventListener('applicationready',
-        this.handleApplicationReady);
+(function() {
+  var AppInstallManager = function() {
+  };
+
+  AppInstallManager.IMPORTS = [
+    'shared/js/homescreens/confirm_dialog_helper.js'
+  ];
+
+  BaseModule.create(AppInstallManager, {
+    name: 'AppInstallManager',
+
+    mapDownloadErrorsToMessage: {
+      'NETWORK_ERROR': 'download-failed',
+      'DOWNLOAD_ERROR': 'download-failed',
+      'MISSING_MANIFEST': 'install-failed',
+      'INVALID_MANIFEST': 'install-failed',
+      'INSTALL_FROM_DENIED': 'install-failed',
+      'INVALID_SECURITY_LEVEL': 'install-failed',
+      'INVALID_PACKAGE': 'install-failed',
+      'APP_CACHE_DOWNLOAD_ERROR': 'download-failed'
+    },
+
+    _start: function() {
+      this.systemBanner = new SystemBanner();
+
+      this.dialog = new AppInstallDialog();
+      this.installCancelDialog = new AppInstallCancelDialog();
+      this.downloadCancelDialog = new AppDownloadCancelDialog();
+      this.setupInstalledAppDialog = new SetupInstalledAppDialog();
+      this.imeLayoutDialog = new ImeLayoutDialog();     
+
+      this.notifContainer =
+              document.getElementById('install-manager-notification-container');
+      this.appInfos = {};
+      this.setupQueue = [];
+      this.isSetupInProgress = false;
+      window.addEventListener('mozChromeEvent',
+        (function ai_handleChromeEvent(e) {
+        if (e.detail.type == 'webapps-ask-install') {
+          this.handleAppInstallPrompt(e.detail);
+        }
+        if (e.detail.type == 'webapps-ask-uninstall') {
+          this.handleAppUninstallPrompt(e.detail);
+        }
+      }).bind(this));
+
+      window.addEventListener('applicationinstall',
+        this.handleApplicationInstall.bind(this));
+
+      window.addEventListener('applicationuninstall',
+        this.handleApplicationUninstall.bind(this));
+
+      this.dialog.elements.installButton.onclick =
+        this.handleInstall.bind(this);
+      this.dialog.elements.cancelButton.onclick =
+        this.showInstallCancelDialog.bind(this);
```

Comment:
Also, if you'd like the manager to show CancelDialog when cancel button is pressed:
```js
this.dialog = new AppInstallDialog(this);
AppInstallDialog = function(manager) {
  this.manager = manager;
};
AppInstallDialog.prototype.handleCancelButton = function() {
  this.manager.showInstallCancelDialog();
};
```